vocabularies/Analytics.md

```diff
@@ -8,38 +8,116 @@ Terms for annotating analytical resources
+[RolledUpPropertyCount](Analytics.xml#L80) *([Experimental](Common.md#Experimental))*|Int16|<a name="RolledUpPropertyCount"></a>Number of properties in the entity instance that have been aggregated away
+[DrillURL](Analytics.xml#L86) *([Experimental](Common.md#Experimental))*|URL|<a name="DrillURL"></a>URL to retrieve more detailed data related to a node of a recursive hierarchy. Annotations with this term MUST include a qualifier to select the hierarchy for which the drill URL is provided.
+[PlanningAction](Analytics.xml#L98) *([Experimental](Common.md#Experimental))*|[Tag](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Core.V1.md#Tag)|<a name="PlanningAction"></a>Processes or generates plan data. Its logic may have side-effects on entity sets.
+[AggregatedProperties](Analytics.xml#L106) *(Deprecated)*|\[[AggregatedPropertyType](#AggregatedPropertyType)\]|<a name="AggregatedProperties"></a>Deprecated in favor of [`AggregatedProperty`](#AggregatedProperty)
+[AggregatedProperty](Analytics.xml#L124)|[AggregatedPropertyType](#AggregatedPropertyType)|<a name="AggregatedProperty"></a>Dynamic property for aggregate expression with specified aggregation method defined on the annotated entity type.
+[AnalyticalContext](Analytics.xml#L144)|\[[AnalyticalContextType](#AnalyticalContextType)\]|<a name="AnalyticalContext"></a>Collection of properties that define an analytical context
+[LevelInformation](Analytics.xml#L307) *([Experimental](Common.md#Experimental))*|[HierarchyType](Hierarchy.md#HierarchyType)|<a name="LevelInformation"></a>Information about grouping levels in the result set of a request including the [`MultiLevelExpand`](#MultiLevelExpand) transformation
+
+
+## Functions
+
+<a name="MultiLevelExpand"></a>
+### [MultiLevelExpand](Analytics.xml#L182) *([Experimental](Common.md#Experimental))*
+
+`$apply` transformation that expands an unnamed leveled hierarchy with custom aggregation of certain properties
+
+Example transformation sequence:
+filter on columns `Industry`, `Amount` and `Currency`,
+order by `Amount` descending,
+show 2 levels, with two exceptions
+```
+$apply=filter(Industry in ('IT','AI'))
+/groupby((Country,Region,Segment,Industry),
+         filter($these/aggregate(Amount) gt 0 and
+                $these/aggregate(Currency) ne null))
+/concat(
+  groupby((Country,Region,Segment,Industry))
+    /aggregate($count as LeavesCount),
+  aggregate(Amount,Currency),
+  Analytics.MultiLevelExpand(
+    Levels=[{"DimensionProperties":["Country"],"AdditionalProperties":["CountryName"]},
+            {"DimensionProperties":["Region"],"AdditionalProperties":["RegionName"]},
+            {"DimensionProperties":["Segment","Industry"],"AdditionalProperties":[]}],
+    Aggregation=["Amount","Currency"],
+    SiblingOrder=[{"Property":"Amount","Descending":true}],
+    ShowLevels=2,
+    ExpandEntries=[{"Entry":["US"],"Levels":0},
```

Comment:
can I omit "Levels" to indicate "Levels":null ?

Reply:
No, because Gateway insists on all properties in a complex type, unless they have a default value. But null cannot be defined as default value, see https://github.com/oasis-tcs/odata-specs/issues/392.